### PR TITLE
disasm/x86.py: minor refactor

### PR DESCRIPTION
**Pull request by disconnect3d**
Minor refactor of the x86 disasm code

### CODE REVIEW

File: pwndbg/gdblib/disasm/x86.py [hunk 1]
```diff
@@ -452,31 +448,26 @@ def _get_syscall_arch_info(self, instruction: PwndbgInstruction) -> Tuple[str, s
         # but only `syscall` and `int 0x80` actually execute syscalls on Linux.
         # So here, we return no syscall name for other instructions and we also
         # handle a case when 32-bit syscalls are executed on x64
-        mnemonic = instruction.mnemonic
-
-        # We read .imm directly, because at this point we haven't enhanced the operands with values
-        is_32bit = mnemonic == "int" and instruction.operands[0].imm == 0x80
-        if not (mnemonic == "syscall" or is_32bit):
-            return (None, None)
+        if (mnemonic := instruction.mnemonic) == "syscall":
```
Comment (disconnect3d) [inline pwndbg/gdblib/disasm/x86.py]:
```suggestion
        mnemonic = instruction.mnemonic
        if mnemonic == "syscall":
```

File: pwndbg/gdblib/disasm/x86.py [hunk 2]
```diff
@@ -453,30 +449,26 @@ def _get_syscall_arch_info(self, instruction: PwndbgInstruction) -> Tuple[str, s
         # So here, we return no syscall name for other instructions and we also
         # handle a case when 32-bit syscalls are executed on x64
         mnemonic = instruction.mnemonic
+        if mnemonic == "syscall":
+            return ("x86-64", "rax")
```
Comment (disconnect3d) [inline pwndbg/gdblib/disasm/x86.py:451]: I believe this is the most common case here

File: pwndbg/gdblib/disasm/x86.py [hunk 3]
```diff
@@ -485,21 +477,24 @@ def memory_string_with_components_resolved(
...
 
             index = pwndbg.gdblib.regs[instruction.cs_insn.reg_name(index)]
-            sz += f"{index}*{scale:#x}"
+            sz += f"{index}*{op.mem.scale:#x}"
             arith = True
 
         if disp != 0:
-            if arith and disp < 0:
-                sz += " - "
-            elif arith and disp >= 0:
```
Comment (disconnect3d) [inline pwndbg/gdblib/disasm/x86.py:500]: Both branches had a condition of `arith == True`

File: pwndbg/gdblib/disasm/x86.py [hunk 4]
```diff
@@ -345,18 +342,19 @@ def _parse_memory(self, instruction: PwndbgInstruction, op: EnhancedOperand, emu
             base = self._read_register(instruction, op.mem.base, emu)
             if base is None:
                 return None
-            target += base
+            target = base
+        else:
+            target = 0
 
         if op.mem.disp != 0:
             target += op.mem.disp
```
Comment (OBarronCS) [inline pwndbg/gdblib/disasm/x86.py]: Come to think of it, we can remove the if statement here.

**Review by OBarronCS — APPROVED** [review]
Looks good - it looks like we still catch all the cases, so this should keep all current functionality intact.

**Pull request merged**